src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,296 @@
+                                                             (get-colors "neutral-5")))]
+        box-style)
+      :align-self "flex-end"
+      :border-color
+      (get-colors "neutral-10"))]]])
+
+(defn render-text
+  [title override-theme &
+   {:keys [typography weight size style]
+    :or   {typography :main-semibold
+           weight     :semi-bold
+           size       :paragraph-1
+           style      (style/title override-theme)}}]
+  [text/text
+   {:typography          typography
+    :accessibility-label :title-name-text
+    :ellipsize-mode      :tail
+    :style               style
+    :override-theme      override-theme
+    :number-of-lines     1
+    :weight              weight
+    :size                size}
+   title])
+
+(defn network-type-text
+  [networkType networkState]
+  (cond
+    (or (= networkState "sending") (= networkState "pending"))      "Pending on"
+    (or (= networkState "confirmed") (= networkState "finalising")) "Confirmed on"
+    (= networkState "finalised")                                    "Finalised on"
+    (= networkState "error")                                        "Failed on"))
+
+(defn steps-text
+  [networkType networkState]
+  (cond
+    (and (= networkType "mainnet")
+         (not= networkState "finalised")
+         (not= networkState "error"))   (str (if (< (@app-state :counter) 4)
+                                               (@app-state :counter)
+                                               "4")
+                                             "/4")
+    (= networkState "finalised")        "Epoch 181,329"
+    (and (= networkType "mainnet")
+         (= networkState "error"))      "0/4"
+    (and (= networkType "optimism/arbitrum")
+         (= networkState "finalising")) "1/1"
+    (= networkType "optimism/arbitrum") "0/1"))
+
+(defn get-status-icon
+  [networkType networkState]
+  (cond
+    (or (= networkState "pending") (= networkState "sending"))      ["pending-state"
+                                                                     (get-colors "neutral-50")]
+    (or (= networkState "confirmed") (= networkState "finalising")) ["positive-state"
+                                                                     (get-colors "success-50")]
+    (= networkState "finalised")                                    ["diamond" (get-colors "success-50")]
+    (= networkState "error")                                        ["negative-state"
+                                                                     (get-colors "danger-50")]))
+
+(defn transaction-progress
+  [{:keys [title
+           on-press
+           accessibility-label
+           networkType
+           networkState
+           container-style
+           override-theme]}]
+  (let [count (reagent/atom 0)]
+    (rn/use-effect
+     (fn []
+       (start-interval networkState)
+       (clear-counter)
+       (fn []
+         (stop-interval)))
+     [networkState])
+    [rn/view
```

Comment:
I think this component should be broken down into smaller ones. There's too much hiccup, and this doesn't perform nearly as well as it could.

One of the big reasons Reagent can help us write more performant components than traditional React apps is because it can quickly verify if subcomponents need to be rendered by doing quick(er) equality checks on the props, since all (most) data structures in Clojure are immutable.

But to take advantage of that, we need to break long chains of hiccup into smaller subcomponents. This will hopefully allow subcomponents to receive only what they need as props, and then Reagent will be able to better prevent React from re-rendering them.

Reply:
@ilmotta can you please explain it more by giving example?
I can can divide the transaction_progress into 3 component based on network - arbitrum, mainnet, optimism. Will that be okay?

Reply:
@teoarjun, you can slice the component tree however you prefer, but the important thing is to slice it so as to minimize the amount re-rendering that can take place right at the Reagent level (before React kicks in).

Normally, we don't need to bother too much with this practice, but this `transaction-progress` component is on the huge side, so it should be sliced.

If you want, you can go beyond these 3 components. For example, the entire tree in the code below could be extracted as well, since it should be re-rendered only when `network-state`, `title`, or `override-theme` change. Therefore, if the `network-type` changes, Reagent will skip re-rendering it.

```clojure
[rn/view {:style style/title-item-container}
 [rn/view {:style style/inner-container}
  [load-icon "placeholder" (colors/theme-colors colors/white colors/neutral-50 override-theme)]
  [rn/view {:style style/title-container}
   [render-text title override-theme]]
  (when (= network-state :error)
    [button/button
     {:size      24
      :icon-left :i/refresh
      :type      :primary}
     "Retry"])]]
```